README.md

```diff
@@ -1,121 +1,131 @@
 This file describes how the tests are defined.
 
-General notes:
-* the directory and file structure exists primarily for organizational purposes and is not relevant to successful execution of the tests (with the exception
-of tests within the schema directory that rely on schema import functionality).
-* when a value of type 'document' needs to be expressed, it is encoded as a string with the `document` annotation.  For example:  `document::"1 2 3"`,
-or as a multi-line string:
-> ```
-> document::'''
->   1
->   2
->   3
-> '''
-> ```
-
-Tests are defined using one of the following forms:
-
-### Type definition test
-To assert expected behavior of one or more constraints, a type definition is provided followed by a list of valid and/or invalid values.
-Each value in the `valid` and `invalid` lists is expected to be valid (or invalid) against the preceeding type definition.
-
-For example:
-> ```
-> type::{
->   type: int,
-> }
-> valid::[-1, 0, 1]       // all valid ints
-> invalid::[a, "b", []]   // none of these are ints
-> ```
-
-### Schema definition test
-To assert expected behavior of types within a schema, a schema definition is provided followed by a list of valid and/or invalid values
-for a specified type from the schema.
-
-For example:
-> ```
-> schema_header::{}
-> type::{
->   name: char,
->   codepoint_length: 1,
-> }
-> type::{
->   name: char_list,
->   type: list,
->   element: char,
-> }
-> schema_footer::{}
-> valid::{
->   char: [a, b, c],                  // list of valid chars
->   char_list: [[d], [e, f, g]],      // list of valid char_lists
-> }
-> invalid::{
->   char: [ab, cd, de],               // list of invalid chars
->   char_list: [[ab], [ab, cd, de]],  // list of invalid char_lists
-> }
-> ```
-
-### Invalid type definition test
-To assert that a type definition should be recognized as invalid, the `invalid_type` annotation is used.
+There are three explicit and two implicit types of test cases.
+
+1. `valid_schema` (implicit) – asserts that a schema document is valid.
+    Every test file should be a valid schema, so this is implied for every schema/file that is present in the test suite.
+2. `invalid_schema` – asserts that a schema document is invalid.
+3. `valid_type` – (implicit) asserts that a given type definition is valid.
+    Every top-level type in the test suite loaded and implicitly tested for validity. 
+4. `invalid_types` – asserts that each type in a given list of type definitions is invalid.
+5. `type/should_match/should_not_match` – tests whether values match or do not match a given type.
+
+Example:
+```ion
+// Asserts that the schema document is a valid Ion Schema
+$ion_schema_2_0
+
+// Asserts that the type "short_string" is valid (because it is declared in the schema).
+type::{
+  name: short_string,
+  type: string,
+  codepoint_length: range::[0, 10],
+}
+
+// A test case that checks to see if certain values match a type
```

Comment:
(Suggestion) Should we add a section heading for these examples rather then comments? It might make it easier to look how a specific type of test works.